### PR TITLE
Enable changelog.com devs to create prod db forks with a single command

### PR DESCRIPTION
**Pull request by gerhard**
> [!WARNING]
> This builds on top of https://github.com/thechangelog/changelog.com/pull/507. Please merge that PR first.

---

## TL;DR

This is the next logical step after migrating to Neon.tech part of:
- https://github.com/thechangelog/changelog.com/pull/492

```console
cd changelog
dagger call db-branch --neon-api-key=env:NEON_API_KEY
```

To learn more, see `changelog/README.md` (added in this PR).

## What else changed?

Part of this, we also deployed a Dagger Engine v0.10.3 on Fly.io so that we don't need any sort of container runtime running locally. I know that @jerodsanto will appreciate this 💪

## Screenshots or it didn't happen

Screenshot 2024-03-28 at 19 37 42

Screenshot 2024-03-28 at 19 38 06

Screenshot 2024-03-28 at 19 32 01

Screenshot 2024-03-28 at 19 32 07

---

Shout-out to @kislerdm for https://github.com/kislerdm/neon-sdk-go

### CODE REVIEW

**Comment by jerodsanto** [comment]
@gerhard 507 is merged. Will you do a quick rebase of this branch so it's easier for me to grok its unique changes?

**Comment by gerhard** [comment]
Done!
